drivers/sensor/bmp388/bmp388.c

```diff
@@ -207,6 +207,102 @@ int bmp388_reg_field_update(const struct device *dev,
 	return cfg->ops->reg_field_update(dev, reg, mask, val);
 }
 
+#if defined(CONFIG_BMP388_VALIDATE_TIMINGS)
+static uint32_t bmp388_pow(uint8_t base, uint8_t power)
+{
+  uint32_t pow_output = 1;
+
+  while (power != 0)
+  {
+    pow_output = base * pow_output;
+    power--;
+  }
+
+  return pow_output;
+}
```

Comment:
It looks like this function is always called with base = 2, so why not use a left shift instead?